lisa/sut_orchestrator/baremetal/platform_.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass, field
+from typing import Any, List, Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import RemoteNode, feature, schema
+from lisa.environment import Environment
+from lisa.platform_ import Platform
+from lisa.sut_orchestrator.baremetal.build.source import Source, SourceSchema
+from lisa.sut_orchestrator.baremetal.cluster.cluster import Cluster, ClusterSchema
+from lisa.sut_orchestrator.baremetal.cluster.ip_getter import IpGetterChecker
+from lisa.sut_orchestrator.baremetal.cluster.readychecker import ReadyChecker
+from lisa.util import check_till_timeout, fields_to_dict
+from lisa.util.logger import Logger
+from lisa.util.shell import try_connect
+from lisa.util.subclasses import Factory
+
+from .. import BAREMETAL
+from . import features
+
+
+@dataclass_json()
+@dataclass
+class SourceBuildSchema:
+    source: Optional[SourceSchema] = None
+
+
+@dataclass_json()
+@dataclass
+class BareMetalPlatformSchema:
+    source: Optional[SourceSchema] = field(default=None)
+    cluster: ClusterSchema = field(default_factory=ClusterSchema)
```

Comment:
As we discussed, the cluster should be an array. We can deal with first one so far, but the schema should be extendable.